functions/internal/javascript/gram-start.mjs

```diff
@@ -37,14 +40,14 @@ class FunctionsError extends Error {
 
 /**
  * @param {string[]} args
- * @returns {{pipePath: string, toolCall: {name: string, input: unknown}}}
+ * @returns {{pipePath: string, request: {name?: string, uri?: string, input: unknown}, type: 'tool' | 'resource'}}
```

Comment:
I think this might be a stricter type:


```suggestion
 * @returns {{
 *   type: "tool",
 *   pipePath: string,
 *   request: { name: string, input: unknown }
 * } | {
 *   type: "resource",
 *   pipePath: string,
 *   request: { uri: string, input: unknown}
 * }}}
```